File: Python/Problems/Problems/problem2785.py
from typing import List

from .problem import Problem


class Problem2785(Problem):
    def test(self) -> bool:
        s = "lEetcOde"

        expected = "lEOtcede"

        result = self.sortVowels(s)

        print(result)

        return result == expected

    def sortVowels(self, s: str) -> str:
        vowelsDic = {
            'A': 0,
            'E': 0,
            'I': 0,
            'O': 0,
            'U': 0,
            'a': 0,
            'e': 0,
            'i': 0,
            'o': 0,
            'u': 0,
        }
        for c in s:
            if self.isVowel(c):
                vowelsDic[c] += 1

        p = 0
        result = list(s)
        for value, count in vowelsDic.items():
            for i in range(count):
                while not self.isVowel(s[p]):
                    p += 1
                result[p] = value
                p += 1

        return ''.join(result)

    def isVowel(self, s: chr) -> bool:
        c = s.lower()
        if c == 'a' or c == 'e' or c == 'i' or c == 'o' or c == 'u':
            return True
        return False

File: C++/Problems/Problems/problem_2785.cpp
#include "problem_2785.h"

#include <map>

bool problem_2785::test() {
    std::string s = "lEetcOde";

    std::string expected = "lEOtcede";

    auto result = sortVowels(s);

    std::cout << result << std::endl;

    return result == expected;
}

std::string problem_2785::sortVowels(std::string s) {
    std::map<char, int> vowelsDic
    {
        {'A', 0},
        {'E', 0},
        {'I', 0},
        {'O', 0},
        {'U', 0},
        {'a', 0},
        {'e', 0},
        {'i', 0},
        {'o', 0},
        {'u', 0},
    };
    for (char c : s)
        if (isVowel(c))
            vowelsDic[c]++;

    int p = 0;
    for (auto vowel : vowelsDic)
    {
        for (int i = 0; i < vowel.second; ++i)
        {
            while (!isVowel(s[p]))
                p++;
            s[p] = vowel.first;
            p++;
        }
    }

    return s;
}


bool problem_2785::isVowel(char c) {
    c = std::tolower(c);
    if (c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u')
        return true;
    return false;
}
